### DISCUSSION TITLE
Personal Access Token (Classic) in Github Action

### DISCUSSION THREAD

**Discussion by alberk8** [community/community#53756] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi,

I have created a PAT (classic) to access github package of a private repository of an Organization.   My question is how do I access or retrieve it into a environment variable of an github action.

The example below does not work, where ACCESS_PACKAGE is the name of the PAT.    The ACCESS_TOKEN is null.

```
env:
  ACCESS_TOKEN: ${{ secrets.ACCESS_PACKAGE }}
```

**Comment by airtower-luna** [comment]
Generally what you quoted should work. If the `ACCESS_TOKEN` environment variable is empty most likely the secret isn't actually available (e.g. because the workflow is running for a PR), but that's hard to say without more context.

**Reply by alberk8** [reply]
If the ACCESS_TOKEN is created in the Secrets section of the Repo then yes, it would work.   What I want to do is to use the PAT directly from the name and not using the token string.   

![image](https://user-images.githubusercontent.com/12510130/234728971-c8c75058-8c25-4666-a4cd-1f0976c57f29.png)

**Reply by airtower-luna** [reply]
That is not possible at all. If you want to have a token available in a workflow, you have to save it as a secret first.

**Reply by alberk8** [reply]
Bummer, It is a good feature to have.   Now I have to create the secret for each repo :(

**Reply by airtower-luna** [reply, marked as answer]
Depending on what you're trying to do, maybe an [organization secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-an-organization) would work? That way you could make secrets available to all repos in the organization.

**Comment by Jekyle944** [comment]
Thx

**Comment by qoomon** [comment]
For the sake of security do not use PAT tokens in github actions. Use my action (or host it on your own) instead https://github.com/qoomon/actions--access-token

**Reply by estahn** [reply]
@laur89 Why do you "down vote" this? It's a legitimate concern.

[a comment by an automated account is not shown]

**Comment by riteshroshann** [comment]
GitHub intentionally doesn’t expose PATs by name at runtime; once created, the token string has to be stored as a secret for it to be injected into a workflow. PATs are not look-up resources, they’re bearer credentials—GitHub won’t let Actions fetch them dynamically because that would make accidental credential leakage trivial. If the inconvenience is replicating the token across multiple repos, org-level secrets or environment secrets are the correct abstraction: a single secret can fan out to many workflows without re-entering the PAT. Long-term, it’s also worth noting that GitHub is steering people away from PAT(Classic) entirely in favor of short-lived OIDC or fine-grained tokens, which eliminates the need to manually distribute long-lived secrets. Out of curiosity, is your use case package read-only access, or do you also need publish rights?
